### ISSUE TITLE
if *expression* unless *expression*

### ISSUE THREAD

**Issue by TrevorBurnham** [jashkenas/coffeescript#956]
Here's a syntactic sugar suggestion (for some post-1.0 version, I suppose): It'd be nice to be able to tack on an `unless` statement to an `if`, e.g.

```
fire() if key is 'space' unless dead or weaponsLocked
```

instead of the current

```
fire() if key is 'space' and not (dead or weaponsLocked)
```

The precedence should be such that you could write

```
run thing if parse orders unless parse exceptions
```

instead of

```
run(thing) if parse(orders) and not parse(exceptions)
```

Interestingly, the statement

```
foo() if a unless b
```

_is_ currently valid, because `a unless b` is a valid statement—one that, for conditional purposes, is equivalent to `a and not b`. There is a difference if `a` or `b` is a function, though, since `if a() unless b()` evaluates `b` first while `if a() and not b()` evaluates `a` first. This is, of course, a pretty fine point. I'm proposing that `if a() unless b()` should be made equivalent to `if a() and not b()`; the current behavior could be achieved by writing `if (a() unless b())`, which I think is somewhat clearer.

Basically, `if a and not b and not c` is a pretty common pattern, and this would make it a bit more readable. Thoughts?
